src/code/z_player_lib.c

```diff
@@ -1187,21 +1921,276 @@ void Player_DrawCouplesMask(PlayState* play, Player* player) {
     AnimatedMat_DrawOpa(play, Lib_SegmentedToVirtual(&object_mask_meoto_Matanimheader_001CD8));
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80127594.s")
+void func_80127594(PlayState* play, Actor* actor) {
+    static Vec3f D_801C0BA8 = { 0.0f, 0.0f, 0.0f };
+    static Vec3f D_801C0BB4 = { 0.0f, 0.0f, 0.0f };
+    Gfx* gfx;
+    s32 i;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    gfx = POLY_XLU_DISP;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801278F8.s")
+    for (i = 0; i < ARRAY_COUNT(D_801C0B90); i++) {
+        f32 temp_f22 = (D_801F59C8[i] / 400.0f) * 0.1f;
+
+        Matrix_MultVec3f(&D_801C0B90[i], &D_801F59B0[i]);
+
+        if (1) {}
+
+        D_801F59B0[i].y += -10.0f * temp_f22;
```

Comment:
```suggestion
        D_801F59B0[i].y -= 10.0f * temp_f22;
```
This probably wouldn't match?

Reply:
No, it doesn't because of float memes